salt/config/__init__.py

```diff
@@ -3331,7 +3331,7 @@ def get_cloud_config_value(name, vm_, opts, default=None, search_global=True):
                 else:
                     value = deepcopy(provider_driver_defs[name])
 
-    if name and vm_ and name in vm_:
+    if name and vm_ and name in vm_ and vm_[name]:
         # The setting name exists in VM configuration.
         if isinstance(vm_[name], types.GeneratorType):
             value = next(vm_[name], '')
```

Comment:
Instead of this, what you should do is

`            if isinstance(value, dict) and isinstance(vm_['name'], dict):`

Because we still want to copy the value over, with a deepcopy in this case.

Reply:
the change goes 2 lines below this diff.